### DISCUSSION TITLE
TextFSM

### DISCUSSION THREAD

**Discussion by amitsova** [ktbyers/netmiko#3357] [Q&A]
Hi everybody,

I am not sure if it is right to open the discussion here, but I assume netmiko users also use TextFSM templates. I am struggling to create a template for the "show interfaces trunk" output, where the highlighted below line have to be matched:

```
**Port        Mode             Encapsulation  Status        Native vlan
Fa1/8       on               802.1q         trunking      1
Gi1/1       on               802.1q         trunking      1
Gi1/2       on               802.1q         trunking      1

Port        Vlans allowed on trunk
Fa1/8       1-4094
Gi1/1       1-4094
Gi1/2       1-4094

Port        Vlans allowed and active in management domain
Fa1/8       1,105,110,130,1010
Gi1/1       1,105,110,130,1010
Gi1/2       1,105,110,130,1010**

Port        Vlans in spanning tree forwarding state and not pruned
Fa1/8       1,105,110,130,1010
Gi1/1       1,105,110,130,1010
Gi1/2       1,105,110,130,1010
```

I would be really thankful if someone can give me a hint or maybe if someone knows a specific TextFSM discussion area.

Thank you in advance!

**Comment by yevgeniy-voloshin** [comment]
Hi Adriana,
maybe this link will shed some light on your problem:
https://stackoverflow.com/questions/50396833/use-textfsm-to-find-allowed-vlans-on-trunk

**Reply by amitsova** [reply]
Hi Yevgeniy!

Thanks a lot! I have something similar:

```
Value PORT (\S+)
Value MODE (\S+)
Value ENCAPS (\S+)
Value STATUS (\S+)
Value NATIVE (\d+)
Value VLAN_ON_TRUNK (\S+)
Value VLAN_MGMT_DOMAIN (\S+)

Start
  ^Port\s+Mode\s+Encapsulation\s+Status\s+Native\s+vlan\s*$$ -> Entry1


  
Entry1
  ^${PORT}\s+${MODE}\s+${ENCAPS}\s+${STATUS}\s+${NATIVE}\s*$$ -> Record
  ^Port\s+Vlans\s+allowed\s+on\s+trunk -> Entry2
  
Entry2
  ^\S+\s+${VLAN_ON_TRUNK}\s*$$ -> Record
  ^Port\s+Vlans\s+allowed\s+and\s+active\s+in\s+management\s+domain\s*$$ -> Entry3
  
Entry3
  ^\S+\s+${VLAN_MGMT_DOMAIN}\s*$$ -> Record
  ^Port\s+Vlans\s+in\s+spanning\s+tree\s+forwarding\s+state\s+and\s+not\s+pruned -> End
```

My problem is that the output looks like this:

![image](https://github.com/ktbyers/netmiko/assets/39456433/6267a59f-e885-4b10-921e-25630f5f81c0)


And I don't know if there is the possibility to have all in one row. Maybe there's a way, but unfortunately, I don't know it.
